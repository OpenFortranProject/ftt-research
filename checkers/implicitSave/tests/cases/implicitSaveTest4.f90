module test
  contains
  function f()
    integer :: x = 4
    print *, x
    x = x + 1
    f = 0
  end function
end module

program main
  use test
  implicit none
  integer :: x
  print *, "Testing f"
  x = f()
  x = f()
  x = f()
end program main
