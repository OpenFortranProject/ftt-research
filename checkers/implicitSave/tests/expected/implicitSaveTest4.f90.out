Warning: environment variable COMPASS_PARAMETERS is not set, trying to use compass_parameters of the current path
 Running Prerequisite SgProject
ImplicitSave: /home/dagit/research/checkers/implicitSave/tests/cases/implicitSaveTest4.f90:6.4: improper operation to variable: expression 'x = x + 1' implicitly saved variable re-assigned
